src/components/CharactersPage/DeleteCharacterModal/DeleteCharacterModal.js

```diff
@@ -27,11 +21,14 @@ const useStyles = makeStyles((theme) => ({
 
 const DeleteCharacterModal = ({ isModalOpen, onClose, character }) => {
   const classes = useStyles()
-
-  const dispatch = useDispatch()
+  const cache = useApolloClient().cache
 
   const handleOnClick = () => {
-    dispatch(deleteCharacter(character.id))
+    cache.evict({
+      id: cache.identify({ __typename: "Person", id: character.id }),
+    })
```

Comment:
This removes the object from the cache entirely, but other queries may still be using that object which will cause errors. You need to be removing the object from the specific query